addons/test_mass_mailing/tests/test_performance.py

```diff
@@ -92,3 +92,6 @@ def test_send_mailing_w_bl(self):
 
         self.assertEqual(mailing.sent, 50)
         self.assertEqual(mailing.delivered, 50)
+
+        cancelled_mail_count = self.env['mail.mail'].sudo().search([('mailing_id', '=', mailing.id)])
+        self.assertEqual(len(cancelled_mail_count), 12, 'Should not have auto deleted the blacklisted emails')
```

Comment:
Are we sure those 12 come from the mailing itself ?

Reply:
Yes I'm pretty sure, because we search on `('mailing_id', '=', mailing.id)`

Reply:
Oh right sorry, did not see the domain. Bad TDE.